### ISSUE TITLE
Cannot add property of type "object" with additionalProperties=true when no_additional_properties is set

### ISSUE THREAD

**Issue by gatopeich** [json-editor/json-editor#1395]
### General information

- json-editor version: 2.10.1

### Expected behavior
Should be able to add any kind of property under a schema with `additionalProperties=true`.
According to the README, even if editor option `no_additional_properties` is set, this should work.
It works for adding simple items, like a list of numbers or a string...

### Actual behavior
But any property of type "object" becomes "null" and unchangeable.

I guess that "no_additional_properties" is applying to the child object which has no schema, so it cannot have `additionalProperties=true`.

### Steps to reproduce the behavior

Direct link to example: https://json-editor.github.io/json-editor/

```js
{
  "title": "json schema example",
  "type": "object",
  "additionalProperties": true,
  "properties": {
    "example": {
      "type": "string",
      "description": "This is an example schema.",
      "default": "Please edit me."
    }
  }
}
```
this object displays correctly:
```js
{"list": [1,2,3]}
```
==> 
![image](https://github.com/json-editor/json-editor/assets/7722268/324f86a1-2e92-45dd-892f-b9c701cea8a8)

But this just displays [null]
```
{"my-object": {"x": 1}}
```
![image](https://github.com/json-editor/json-editor/assets/7722268/4404e4ee-dd1a-4ce7-b1e6-3b17e06e3fa7)
```

**Comment by StephanSchuster** [comment]
I have a similar issue. But what you are describing here works fine for me. Maybe you want so check my issue: https://github.com/json-editor/json-editor/issues/1410

**Comment by germanbisurgi** [comment]
fixed in #1415
